pageserver/src/walreceiver/connection_manager.rs

```diff
@@ -90,19 +91,32 @@ pub fn spawn_connection_manager_task(
             }
         }
         .instrument(
-            info_span!(parent: None, "wal_connection_manager", tenant = %tenant_id, timeline = %timeline_id),
+            info_span!(parent: None, "wal_connection_manager", tenant = %id.tenant_id, timeline = %id.timeline_id),
         ),
     );
 }
 
+macro_rules! acquire_timeline_read {
```

Comment:
Like with `PageReconstructResult`, this macro here is a substitute for `std::ops::Try`.
IMO the name should contain `try` to make it clear at the call site that the macro returns if we can't upgrade the Weak to an Arc.